### PR TITLE
Bot detection Feature - APIM 3.0.0 - product changes

### PR DESCRIPTION
**Pull request by nadeepoornima**
Issue
[1]. wso2/product-apim#5358

Have changed the log4j and main.xml file relating to Bot detection feature.

### CODE REVIEW

File: modules/distribution/product/src/main/conf/log4j2.properties [hunk 1]
```diff
@@ -398,3 +398,14 @@ rootLogger.appenderRef.ERROR_LOGFILE.ref = ERROR_LOGFILE
 rootLogger.appenderRef.PaxOsgi.ref = PaxOsgi
 #rootLogger.appenderReg.CARBON_SYS_LOG.ref = CARBON_SYS_LOG
 #rootLogger.appenderRef.syslog.ref = syslog
+
+#Appender for Bot Data Detection
+appender.BOTDATA_APPENDER=org.apache.log4j.RollingFileAppender
```
Comment (tharindu1st) [inline modules/distribution/product/src/main/conf/log4j2.properties]: did we add this appender to top

Comment (nadeepoornima) [reply modules/distribution/product/src/main/conf/log4j2.properties]: Added

File: modules/distribution/product/src/main/conf/log4j2.properties [hunk 2]
```diff
@@ -398,3 +398,14 @@ rootLogger.appenderRef.ERROR_LOGFILE.ref = ERROR_LOGFILE
...
+appender.BOTDATA_APPENDER.File=${carbon.home}/repository/logs/${instance.log}/wso2-BotDetectedData${instance.log}.log
+appender.BOTDATA_APPENDER.MaxFileSize=1000KB
+appender.BOTDATA_APPENDER.MaxBackupIndex=10
+appender.BOTDATA_APPENDER.layout=org.apache.log4j.PatternLayout
+appender.BOTDATA_APPENDER.layout.ConversionPattern=%d{ISO8601} [%X{ip}-%X{host}] [%t] %5p %c{1} %m%n 
+logger.org.wso2.carbon.apimgt.gateway.mediators.BotDetectionMediator=INFO, BOTDATA_APPENDER
```
Comment (tharindu1st) [inline modules/distribution/product/src/main/conf/log4j2.properties]: This need to be put on log4j2 way

File: modules/distribution/product/src/main/conf/synapse-configs/default/sequences/main.xml [hunk 3]
```diff
@@ -24,7 +24,24 @@
         <log level="custom">
             <property name="STATUS" value="Message dispatched to the main sequence. Invalid URL."/>
             <property name="RESOURCE" expression="fn:concat('/', $axis2:REST_URL_POSTFIX)"/>
-        </log>        
+            <property name="HEALTH CHECK URL" expression="$axis2:TransportInURL"/>
+        </log>
+        <property name="url_axis2" expression="$axis2:TransportInURL"/>
```
Comment (bhathiya) [inline modules/distribution/product/src/main/conf/synapse-configs/default/sequences/main.xml]: do we use this property?

**Pull request merged**